src/main/java/fi/solita/clamav/ClamAVProxy.java

```diff
@@ -46,9 +52,47 @@ public String ping() throws IOException {
   @RequestMapping(value="/scanReply", method=RequestMethod.POST)
   public @ResponseBody String handleFileUploadReply(@RequestParam("name") String name,
                                                     @RequestParam("file") MultipartFile file) throws IOException{
+    return sendToClamAV(name, file, true);
+  }
+
+  /**
+   *
+   * @param name The filename with extension
+   * @param file The file content
+   * @param returnClamAVResponse If true, will return the full clamav response, else will return a string containing success or failure
+   * @return The appropriate response based on whether `returnClamAVResponse` is specified
+   * @throws IOException
+   */
+  private String sendToClamAV(String name, MultipartFile file, boolean returnClamAVResponse) throws IOException {
     if (!file.isEmpty()) {
-      ClamAVClient a = clamAVClientFactory.newClient();
-      return new String(a.scan(file.getInputStream()));
+      //Write to disk
+      Path tempFilePath = Files.createTempFile("unsafe", name);
+      try {
+        file.transferTo(tempFilePath);
+
+        //Validate from disk
+        try(InputStream stream = Files.newInputStream(tempFilePath)) {
+          BufferedInputStream validatorStream = new BufferedInputStream(stream);
+          MediaTypeValidator.verifyFileIsOfAcceptedType(validatorStream, name);
+        }
+
+        //Scan from disk
+        try(InputStream stream = Files.newInputStream(tempFilePath)) {
+          BufferedInputStream scannerStream = new BufferedInputStream(stream);
```

Comment:
```suggestion
        try(InputStream stream = Files.newInputStream(tempFilePath);
          BufferedInputStream scannerStream = new BufferedInputStream(stream)) {
```
Not sure if this makes much difference but you should be able to put both streams inside the try.

Reply:
No longer relevant as the TikaInputStream is directly opened within the validator.

Reply:
These are separate streams than what you're using with TIka.

Reply:
Missed that, apologies, the change is now [here](https://github.com/Accurx/clamav-rest/commit/116441f4d47924a16a92b189032da9afc414bcd2)